src/app/services/actions/samples-action-creators.ts

```diff
@@ -55,7 +57,11 @@ export function fetchSamples() {
       const res = await response.json();
       return dispatch(fetchSamplesSuccess(res.sampleQueries));
     } catch (error) {
-      return dispatch(fetchSamplesError({ error }));
+      let cachedSamples = await samplesCache.readSamples();
+      if (cachedSamples.length === 0) {
```

Comment:
Can we check if cachedSamples are available before accessing their .length property?
if(cachedSamples && cachedSamples.length === 0)

Reply:
It always returns a list. An empty list when there's nothing to show and an actual list when content exists